pkg/daemon/ceph/client/pool.go

```diff
@@ -326,18 +336,112 @@ func CreateReplicatedPoolForApp(context *clusterd.Context, clusterInfo *ClusterI
 	return nil
 }
 
-func createReplicationCrushRule(context *clusterd.Context, clusterInfo *ClusterInfo, ruleName string, pool cephv1.PoolSpec) error {
-	failureDomain := pool.FailureDomain
-	if failureDomain == "" {
-		failureDomain = cephv1.DefaultFailureDomain
+func createStretchedReplicationCrushRule(context *clusterd.Context, clusterInfo *ClusterInfo, ruleName string, pool cephv1.PoolSpec) error {
+	// Get the current CRUSH map
+	var crushMap CrushMap
+	crushMap, err := GetCrushMap(context, clusterInfo)
+	if err != nil {
+		return errors.Wrap(err, "failed to get crush map")
+	}
+
+	// Fetch the compiled crush map
+	compiledCRUSHMapFilePath, err := GetCompiledCrushMap(context, clusterInfo)
+	if err != nil {
+		return errors.Wrap(err, "failed to get crush map")
+	}
+
+	// Decompile the plain text to CRUSH binary format
+	err = decompileCRUSHMap(context, compiledCRUSHMapFilePath)
+	if err != nil {
+		return errors.Wrap(err, "failed to compile crush map")
+	}
+
+	// Build plain text rule
+	plainRule := buildStretchClusterPlainCrushRule(crushMap, ruleName, pool)
+
+	// Append plain rule to the decompiled crush map
+	decompiledCRUSHMapFilePath := buildDecompileCRUSHFileName(compiledCRUSHMapFilePath)
+	f, err := os.OpenFile(filepath.Clean(decompiledCRUSHMapFilePath), os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0400)
+	if err != nil {
+		return errors.Wrapf(err, "failed to open decompiled crush map %q", decompiledCRUSHMapFilePath)
+	}
+	defer func() {
```

Comment:
Do we need to move the defer definition sooner? Otherwise a failure above might leave the temp files, right? Or else have a separate defer for each file.